index/scorch/merge.go

```diff
@@ -159,6 +159,20 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
 	// process tasks in serial for now
 	var notifications []chan *IndexSnapshot
 	var filenames []string
+	// clean up any pending notifications from introducer on exit
+	// from an index closure.
+	defer func() {
+		for _, notification := range notifications {
+			if len(notification) > 0 {
```

Comment:
imho, using len(ch) ... (or, as an aside, the related cap(ch))... is a bit of a worrisome no-no in golang, usually indicating a troublesome area...

...because in-between checking the len(ch) and the reading of that ch, theoretically, some other goroutine might sneak in and change the channel's state.  Now, that may not be the case here...

but instead, maybe use the select { case <-ch ...; default } idiom instead?